stl/inc/utility

```diff
@@ -750,6 +750,10 @@ template <class _Ty>
 _NODISCARD constexpr underlying_type_t<_Ty> to_underlying(_Ty _Value) noexcept {
     return static_cast<underlying_type_t<_Ty>>(_Value);
 }
+
+__declspec(noreturn) void unreachable() {
+    __assume(false);

```

Comment:
We can, but it does not matter I think.

Reply:
tried out `clang-cl`. `__assume(false)` works, and either `__assume(false)` or `[[noreturn]]` alone is enough.

Reply:
I'll ask in Discord, and come back if Clang devs suggest we should prefer `__builtin_unreachable`.

Reply:
Reopening: `__builtin_unreachable` is apparently infinitesimally better since (1) `-mllvm -trap-unreachable` will emit trap instructions for `__builtin_unreachable` and (2) ubsan understands it (this may matter to us someday. (See https://discord.com/channels/636084430946959380/636732781086638081/940384796952133662)

I think this is just motivating enough to throw in a preprocessor conditional here.